src/helpers/proofs.js

```diff
@@ -0,0 +1,237 @@
+    String(encodedPartialPath[0]) === "2"
+  ) {
+    partialPath = encodedPartialPath.slice(2)
+  } else {
+    partialPath = encodedPartialPath.slice(1)
+  }
+
+  if (partialPath === path.slice(pathPtr, pathPtr + partialPath.length)) {
+    return partialPath.length
+  } else {
+    throw new Error("path was wrong")
+  }
+}
+
+export function getTxBytes(tx) {
+  const txObj = new EthereumTx(squanchTx(tx))
+  return txObj.serialize()
+}
+
+// build
+export async function getTxProof(tx, block) {
+  const txTrie = new Trie()
+  for (let i = 0; i < block.transactions.length; i++) {
+    const siblingTx = block.transactions[i]
+    const path = rlp.encode(siblingTx.transactionIndex)
+    const rawSignedSiblingTx = getTxBytes(siblingTx)
+    await new Promise((resolve, reject) => {
+      txTrie.put(path, rawSignedSiblingTx, err => {
+        if (err) {
+          reject(err)
+        } else {
+          resolve()
+        }
+      })
+    })
+  }
+
+  // promise
+  return new Promise((resolve, reject) => {
+    txTrie.findPath(
+      rlp.encode(tx.transactionIndex),
+      (err, rawTxNode, reminder, stack) => {
+        if (err) {
+          return reject(err)
+        }
+
+        if (reminder.length > 0) {
+          return reject(new Error("Node does not contain the key"))
+        }
+
+        const prf = {
+          blockHash: utils.toBuffer(tx.blockHash),
+          parentNodes: stack.map(s => s.raw),
+          root: getRawHeader(block).transactionsTrie,
+          path: rlp.encode(tx.transactionIndex),
+          value: rlp.decode(rawTxNode.value)
+        }
+        resolve(prf)
+      }
+    )
+  })
+}
+
+export function verifyTxProof(proof) {
+  const path = proof.path.toString("hex")
+  const value = proof.value
+  const parentNodes = proof.parentNodes
+  const txRoot = proof.root
+  try {
+    var currentNode
+    var len = parentNodes.length
+    var nodeKey = txRoot
+    var pathPtr = 0
+    for (var i = 0; i < len; i++) {
+      currentNode = parentNodes[i]
+      const encodedNode = Buffer.from(
+        utils.sha3(rlp.encode(currentNode)),
+        "hex"
+      )
+      if (!nodeKey.equals(encodedNode)) {
+        return false
+      }
+      if (pathPtr > path.length) {
+        return false
+      }
+      switch (currentNode.length) {
+        case 17: // branch node
+          if (pathPtr === path.length) {
+            if (currentNode[16] === rlp.encode(value)) {
+              return true
+            } else {
+              return false
+            }
+          }
+          nodeKey = currentNode[parseInt(path[pathPtr], 16)] // must === sha3(rlp.encode(currentNode[path[pathptr]]))
+          pathPtr += 1
+          break
+        case 2:
+          pathPtr += nibblesToTraverse(
+            currentNode[0].toString("hex"),
+            path,
+            pathPtr
+          )
+          if (pathPtr === path.length) {
+            // leaf node
+            if (currentNode[1].equals(rlp.encode(value))) {
```

Comment:
here as well `return currentNode[1].equals(rlp.encode(value)`
or if it's not boolean then `return !!currentNode[1].equals(rlp.encode(value)`